tests/src/core/geometry/testqgscurvepolygon.cpp

```diff
@@ -1705,6 +1709,48 @@ void TestQgsCurvePolygon::testCast()
   QVERIFY( !pCast2.fromWkt( QStringLiteral( "CurvePolygonZ((111111))" ) ) );
 }
 
+void TestQgsCurvePolygon::testArea3D()
+{
+  QgsCurvePolygon poly;
+
+  std::unique_ptr<QgsCircularString> ext = std::make_unique<QgsCircularString>();
```

Comment:
same suggestion as above. Test cases with a unit square on planes x = 0 and y = 0 would be great

Reply:
and also testing a "U" shape polygon with a mix of convexity and concavity, like (0 0,0 2,1 2,1 1,2 1,2 2,2 0,0 0))